async_upnp_client/ssdp_listener.py

```diff
@@ -402,22 +398,18 @@ def see_advertisement(
             notification_sub_type == NotificationSubType.SSDP_UPDATE
             or is_new_device
             or is_new_service
+            or new_location
             or headers_differ_from_existing_advertisement(
                 ssdp_device, notification_type, headers
             )
-            or headers_differ_from_existing_search(
-                ssdp_device, notification_type, headers
-            )
-            or new_location
         )
 
         # Update stored headers.
-        if notification_type in ssdp_device.advertisement_headers:
-            ssdp_device.advertisement_headers[notification_type].replace(headers)
+        advertisement_headers = ssdp_device.advertisement_headers
+        if notification_type in advertisement_headers:
+            advertisement_headers[notification_type].replace(headers)
         else:
-            ssdp_device.advertisement_headers[notification_type] = CaseInsensitiveDict(
-                headers
-            )
+            advertisement_headers[notification_type] = headers
```

Comment:
At one point this was typed to have something else than `CaseInsensitiveDict` or not typed at all. This now matches the code on the search side